### PR TITLE
Applicative cleanup

### PR DESCRIPTION
**Pull request by MonoidMusician**
I feel strongly about `fold` being in temporal order.

`sampleOn` having the same signature as `<*>` actually seems like more of a hassle than it's worth, especially in the `Behavior` side of things. idk

### CODE REVIEW

**Review by mikesol — COMMENTED** [review]
If `fold` has the same signature as `foldl` now, can we make `AnEvent m` an instance or `Foldable` (and also possibly `Traversable`) and drop fold from `IsEvent`?

**Comment by mikesol** [comment]
> `sampleOn` having the same signature as `<*>` actually seems like more of a hassle than it's worth, especially in the `Behavior` side of things. idk

Yeah it requires lots of rewriting.
But the nice thing about the same signature is the fixity - you can fill a record with infix operators.

On that subject, if we change the order, there should be a `sampleOnWithFlippedTime` that takes the temporality from the other event so that, when constructing a long chain with infix operators, one can control which side controls ⏳ .

**Comment by mikesol** [comment]
My gut feeling is that the `sampleOn` signature is the way it is because it allows you to get a `Behavior` out of an `Event` by writing `sampleOn e`. IIRC, all the `Event` stuff grew as an appendage to `Behavior` and eventually took a life of its own, so it makes sense that the signatures are behavior-friendly.

In production workloads, though, it's way more likely that you'll see:

```purescript
{ a: _, b: _, c: _, d: } <$> eventA <*> eventB <|* eventC <*> eventD
```

than making a behavior out of `sampleOn`, so I'd optimize for the record-making case.

**Comment by MonoidMusician** [comment]
It can't be an instance of `Foldable`: that requires you to return a pure value `b` instead of `event b`.

**Comment by mikesol** [comment]
> It can't be an instance of `Foldable`: that requires you to return a pure value `b` instead of `event b`.

Argfl yes this is true 😞

**Comment by mikesol** [comment]
I used the GH editor for a first pass, I'm guessing it'll break but I'll tweak it till it builds & passes tests!

**Comment by MonoidMusician** [comment]
I think you need to change the operators `<|**>`, `<**|>`, `<|*>`, and `<*|>` now, the intent was that single `*` = “same order as `<*>` (function on left)” and double `**` = “opposite order (function on right)”.

File: src/FRP/Event.purs [hunk 1]
```diff
@@ -211,13 +211,13 @@ sampleOnLeft :: forall a b. Event (a -> b) -> Event a ->  Event b
...
+    c2 <-
+      runEffectFn2 e2 b $ mkEffectFn1 \a -> do
+        Ref.write (Just a) latest
     c1 <-
       runEffectFn2 e1 b $ mkEffectFn1 \f -> do
         o <- Ref.read latest
         for_ o (\a -> runEffectFn1 k (f a))
-    c2 <-
-      runEffectFn2 e2 b $ mkEffectFn1 \a -> do
-        Ref.write (Just a) latest
```
Comment (MonoidMusician) [inline src/FRP/Event.purs]: Why did you move this? We want to subscribe to `e1` first, for the cases in which subscription order matters, otherwise it's just `flip sampleOnRightOp` or whatever.

Comment (mikesol) [reply src/FRP/Event.purs]: Ah ok got it now, I'll add some docs to this effect.
